### ISSUE TITLE
Logic Error in gradient accumulation?

### ISSUE THREAD

**Issue by bdebowski** [karpathy/nanoGPT#206]
Line 293 in train.py performs loss.backward() without first normalizing the loss value to account for number of gradient accumulation steps.  I.e if I have 10 gradient accumulation steps I would want to divide loss by 10 before applying backward().  

I'm using https://github.com/google-research/xtreme/blob/master/third_party/run_classify.py as reference material.  (See line 210 or search for "loss = loss / args.gradient_accumulation_steps")

**Comment by chrisociepa** [comment]
It was already discussed in https://github.com/karpathy/nanoGPT/issues/168

**Comment by bdebowski** [comment]
Closing since this duplicates concerns raised in #168

**Comment by bdebowski** [comment]
Closing as this duplicates concerns raised in #168
